src/Mvc/Mvc.Core/src/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -273,6 +273,7 @@ internal static void AddMvcCoreServices(IServiceCollection services)
             services.TryAddSingleton<OrderedEndpointsSequenceProviderCache>();
             services.TryAddSingleton<ControllerActionEndpointDataSourceIdProvider>();
             services.TryAddSingleton<ActionEndpointFactory>();
+            services.TryAddSingleton<IRequestDelegateFactory, ControllerRequestDelegateFactory>();
```

Comment:
Could you include the commit for `IRequestDelegateFactory` in this change? It seems sounds, but like you mentioned, we should put the use of this behind a compat flag.